src/map/ai/helpers/targetfind.cpp

```diff
@@ -458,6 +458,24 @@ bool CTargetFind::validEntity(CBattleEntity* PTarget)
         return false;
     }
 
+    if (PTarget->objtype == TYPE_MOB)
+    {
+        if (m_PBattleEntity &&
+            m_PBattleEntity->objtype == TYPE_PC &&
+            !static_cast<CCharEntity*>(m_PBattleEntity)->IsMobOwner(PTarget))
```

Comment:
dynamic_cast would be safer since we're casting downwards in the hierarchy (although I can't think of why this would fail if we've already confirmed TYPE_PC).

Reply:
The reason dynamic wasn't used here was for performance in this case. Dynamic vs static is a fairly significant performance difference and if on a crowded cluster it could make a difference if there are a lot of aoes going off at the same time.